### PR TITLE
Remove ability to increase authority

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

File: sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala [hunk 1]
```diff
@@ -1081,12 +1081,8 @@ private[lf] object SBuiltinFun {
...
             doExe(Some(authorizers))
           } else {
-            // additional authority *is* required; ask the ledger
-            machine.needAuthority(
-              NameOf.qualifiedNameOfCurrentFunc,
-              holding = holding,
-              requesting = requesting,
-              continue = () => doExe(Some(authorizers)),
+            Control.Error(
+              IE.RejectedAuthorityRequest(holding = holding, requesting = requesting)
```
Comment (samuel-williams-da) [inline sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala:1085]: So we're still allowing authority to be reduced, which is why this error still exists? Verses just using `FailedAuthorization`

File: sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/ChoiceAuthorityTest.scala [hunk 2]
```diff
@@ -200,15 +185,10 @@ class ChoiceAuthorityTest(majorLanguageVersion: LanguageMajorVersion)
...
-        inside(err) { case SError.SErrorDamlException(FailedAuthorization(_, why)) =>
-          inside(why) { case cma: CreateMissingAuthorization =>
-            cma.authorizingParties shouldBe Set(charlie)
-            cma.requiredParties shouldBe Set(alice)
-          }
-        }
-      }
+    "try to gain authority {A,B}-->{C} (need A)" in {
+      runExample(theAut = Set(charlie), theGoal = alice) shouldBe Left(
+        SError.SErrorDamlException(RejectedAuthorityRequest(Set(alice, bob), Set(charlie)))
```
Comment (samuel-williams-da) [inline sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/ChoiceAuthorityTest.scala:190]: This exception changed? Should `RejectedAuthorityRequest` not be inside a `FailedAuthorization`?

**Review by samuel-williams-da — APPROVED** [review]
Small notes

**Comment by samuel-williams-da** [comment]
Why is this happening by the way? Wasn't this feature added for consortium parties?

File: sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala:1085]: > So we're still allowing authority to be reduced, which is why this error still exists?

**Comment by remyhaemmerle-da** [comment]
> Why is this happening by the way? Wasn't this feature added for consortium parties?
1. the feature was never implemented on the canton side CN does not need the features anymore

File: sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala]: Finally I simplify the exception. 
I just used FailedAuthorization, i.e.  we use this exception when the `controllers union  choiceAuthorizer` are not a subset of the `authorized parties`.

With this change I could simplify a bit the BeginExercise.

**Review by samuel-williams-da — APPROVED** [review]
Looks good! Do we need to remove any handling of this error on the canton side, or are we all good?

File: sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala [hunk 3]
```diff
@@ -1035,59 +1035,36 @@ private[lf] object SBuiltinFun {
         )
         val obsrs = extractParties(NameOf.qualifiedNameOfCurrentFunc, args.get(3))
         machine.enforceChoiceObserversLimit(obsrs, coid, templateId, choiceId, chosenValue)
-        val authorizersWhenExplicit =
-          extractParties(NameOf.qualifiedNameOfCurrentFunc, args.get(4))
-        machine.enforceChoiceAuthorizersLimit(
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala:1040]: No ! 
Good catch.

[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala:1040]: add missing test.

**Pull request merged**